extensions/amp-analytics/0.1/test/test-linker-manager.js

```diff
@@ -308,6 +308,57 @@ describe('Linker Manager', () => {
     });
   });
 
+  it('should match friendly domain if destinationDomains unspecified', () => {
+    const config = {
+      linkers: {
+        enabled: true,
+        testLinker1: {
+          ids: {
+            id: '111',
+          },
+        },
+      },
+    };
+
+    const manager = new LinkerManager(ampdoc, config);
+    sandbox.stub(manager, 'expandTemplateWithUrlParams_').returnsArg(0);
+    manager.init();
+
+    return Promise.all(manager.allLinkerPromises_).then(() => {
+      const url1 = {
+        href: 'https://www.source.com/path',
+        hostname: 'www.source.com',
+      };
+      const url2 = {
```

Comment:
I'd rewrite this test to have a helper function and then test one link at a time and add a few more test cases like

`function checkLinkApplied('https://www.source.com/path') {`

Reply:
done. all tests rewritten, less mocks more real tests.